app/graph/mutations/team_mutations.rb

```diff
@@ -1,39 +1,60 @@
-
-  settings_fields = {
-    slack_notifications_enabled: 'str',
-    slack_webhook: 'str',
-    slack_notifications: 'str',
-    language: 'str',
-    languages: 'json',
-    language_detection: 'bool',
-    list_columns: 'json',
-    tipline_inbox_filters: 'str',
-    suggested_matches_filters: 'str',
-    outgoing_urls_utm_code: 'str',
-    shorten_outgoing_urls: 'bool'
-  }
-
-  update_fields = fields.merge(settings_fields).merge({
-    name: 'str',
-    add_auto_task: 'json',
-    media_verification_statuses: 'json',
-    set_team_tasks: 'json',
-    rules: 'str',
-    remove_auto_task: 'str', # label
-    empty_trash: 'int',
-    report: 'json',
-  })
-
-  Create, Update, Destroy = GraphqlCrudOperations.define_crud_operations('team', create_fields, update_fields, ['check_search_team', 'check_search_trash', 'check_search_spam', 'check_search_unconfirmed', 'public_team'])
+  MUTATION_TARGET = 'team'.freeze
+  PARENTS = [
+    'public_team',
+    # TODO: consolidate parent class logic if present elsewhere
+    { check_search_team: CheckSearchType },
+    { check_search_trash: CheckSearchType },
+    { check_search_spam: CheckSearchType },
+    { check_search_unconfirmed: CheckSearchType },
+  ].freeze
+
+  module SharedCreateAndUpdateFields
+    extend ActiveSupport::Concern
+
+    included do
+      argument :archived, Integer, required: false
+      argument :private, GraphQL::Types::Boolean, required: false
+      argument :description, String, required: false
+
+      # TODO: extract as TeamAttributes module
+      field :team_userEdge, TeamUserType.edge_type, camelize: false, null: true
+      field :user, UserType, null: true
+    end
+  end
+
+  class Create < Mutations::CreateMutation
+    include SharedCreateAndUpdateFields
+
+    argument :name, String, required: true
+    argument :slug, String, required: true
+  end
+
+  class Update < Mutations::UpdateMutation
+    include SharedCreateAndUpdateFields
+
+    argument :name, String, required: false
+    argument :add_auto_task, JsonString, required: false, camelize: false
+    argument :media_verification_statuses, JsonString, required: false, camelize: false
+    argument :set_team_tasks, JsonString, required: false, camelize: false
+    argument :rules, String, required: false
+    argument :remove_auto_task, String, required: false, camelize: false # label
+    argument :empty_trash, Integer, required: false, camelize: false
+    argument :report, JsonString, required: false
+
+    # Settings fields
+    argument :slack_notifications_enabled, String, required: false, camelize: false
+    argument :slack_webhook, String, required: false, camelize: false
+    argument :slack_notifications, String, required: false, camelize: false
+    argument :language, String, required: false
+    argument :languages, JsonString, required: false
+    argument :language_detection, Boolean, required: false, camelize: false
+    argument :list_columns, JsonString, required: false, camelize: false
```

Comment:
It's not! Or, well, it might be (there's a [JSON type]([GraphQL](https://graphql-ruby.org/api-doc/2.0.23/GraphQL.html)::[Types](https://graphql-ruby.org/api-doc/2.0.23/GraphQL/Types.html)::JSON) at least in 2.0), but we're not using it. We still define it ourselves, I just renamed the class from `JsonStringStype` to `JsonString` because it was more in pattern with the other scalar object types

Reply:
No, it's clear... the source of confusion to me was not even the rename, just the fact that they don't differentiate between custom types and core types... maybe in the future we can namespace our custom types for clarity, but not needed for now (it can go under a "nice to have in the future" list).

Reply:
Yeah, I agree that would be nice. We could just reference the base types by full name (ie GraphQL::Types::X) right now. I had tried to do some differentiation via module at the start but as you saw when we paired it was kind of a mess. Would be great as a refactor.

For now I might go ahead and just do a find and replace to declare the GraphQL base types by their full name. I had some other questions (like whether we're actually referencing the Ruby String class in included modules rather than the GraphQL ones) that changing the name would resolve, too, and it's an easy fix.